multiversx_sdk_cli/contracts.py

```diff
@@ -61,161 +64,142 @@ class IContractQueryResponse(Protocol):
     return_message: str
 
 
-class SmartContract:
-    def __init__(self, address: Optional[IAddress] = EmptyAddress(), bytecode=None, metadata=None):
-        self.address = address
-        self.bytecode = bytecode
-        self.metadata = metadata or CodeMetadata()
-
-    def deploy(self, owner: Account, arguments: List[Any], gas_price: int, gas_limit: int, value: int, chain: str, version: int, guardian: str, options: int) -> Transaction:
-        self.owner = owner
-        address_computer = AddressComputer(number_of_shards=3)
-        self.address = address_computer.compute_contract_address(self.owner.address, self.owner.nonce)
-
-        arguments = arguments or []
-        gas_price = int(gas_price)
-        gas_limit = int(gas_limit)
-        value = value or 0
-
-        tx = Transaction(
-            chain_id=chain,
-            sender=owner.address.to_bech32(),
-            receiver=ADDRESS_ZERO_BECH32,
-            gas_limit=gas_limit,
-            gas_price=gas_price,
-            nonce=owner.nonce,
-            amount=value,
-            data=self.prepare_deploy_transaction_data(arguments).data,
-            version=version,
-            options=options
-        )
+class IConfig(Protocol):
+    chain_id: str
+    min_gas_limit: int
```

Comment:
from what i've seen IConfig and ITokenComputer are defined also in core, i'm thinking if it might be better to reuse from core